src/backend/joanie/payment/api.py

```diff
@@ -67,7 +67,12 @@ def get_queryset(self):
             if self.request.auth
             else self.request.user.username
         )
-        return models.CreditCard.objects.filter(owner__username=username)
+
+        payment_provider = get_payment_backend()
+
+        return models.CreditCard.objects.filter(
+            owner__username=username, payment_provider=payment_provider.name
+        )
```

Comment:
Should we not use `get_card_for_owner` here ?

Reply:
We may use the manager method and do some little adjustment to fit the pk as an kwarg to add.
If we add **kwargs to the parameters of the method, it will allow to customize the filter of the queryset.